contracts/NodeDelegator.sol

```diff
@@ -145,82 +153,121 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
+    /// This includes LSTs and WETH.
+    /// @return assets a list of assets addresses
+    /// @return assetBalances the balances of the assets in the node delegator and EigenLayer
     function getAssetBalances()
         external
         view
         override
         returns (address[] memory assets, uint256[] memory assetBalances)
     {
-        address eigenlayerStrategyManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_STRATEGY_MANAGER);
+        assets = lrtConfig.getSupportedAssetList();
+        assetBalances = new uint256[](assets.length);
 
-        (IStrategy[] memory strategies,) =
-            IEigenStrategyManager(eigenlayerStrategyManagerAddress).getDeposits(address(this));
+        for (uint256 i = 0; i < assets.length;) {
+            (uint256 assetLyingInNDC, uint256 assetStakedInEigenLayer) = getAssetBalance(assets[i]);
+            assetBalances[i] = assetLyingInNDC + assetStakedInEigenLayer;
 
-        uint256 strategiesLength = strategies.length;
-        assets = new address[](strategiesLength);
-        assetBalances = new uint256[](strategiesLength);
-
-        for (uint256 i = 0; i < strategiesLength;) {
-            assets[i] = address(IStrategy(strategies[i]).underlyingToken());
-            assetBalances[i] = IStrategy(strategies[i]).userUnderlyingView(address(this));
             unchecked {
                 ++i;
             }
         }
     }
 
-    /// @dev Returns the balance of an asset that the node delegator has deposited into the strategy
-    /// @param asset the asset to get the balance of
-    /// @return stakedBalance the balance of the asset
-    function getAssetBalance(address asset) external view override returns (uint256) {
-        address strategy = lrtConfig.assetStrategy(asset);
-        if (strategy == address(0)) {
-            return 0;
-        }
+    /// @dev Returns the balance of an asset that the node delegator has deposited into an EigenLayer strategy
+    /// or native ETH staked into an EigenPod.
+    /// @param asset the token address of the asset.
+    /// WETH will include any native ETH in this contract or staked in EigenLayer.
+    /// @return assetLyingInNDC assets lying in this NDC contract.
+    /// This includes any native ETH when the asset is WETH.
+    /// @return assetStakedInEigenLayer asset amount deposited in underlying EigenLayer strategy
+    /// or native ETH staked into an EigenPod.
+    function getAssetBalance(address asset)
+        public
+        view
+        override
+        returns (uint256 assetLyingInNDC, uint256 assetStakedInEigenLayer)
+    {
+        assetLyingInNDC += IERC20(asset).balanceOf(address(this));
+
+        // The WETH address is different between chains so reading from the config.
+        // Ideally, the WETH address would be an immutable but following the existing pattern of using config for now.
+        address WETH_TOKEN_ADDRESS = lrtConfig.getLSTToken(LRTConstants.WETH_TOKEN);
```

Comment:
We could just make WETH a storage slot, since it's used all over the place, and when used in this contract doesn't overlap the existing asset system. Just removeing the external calls would reduce boilerplate and reduce gas far more than all the unchecked `++I`'s in this code.

We can pull it from the `lrtConfig` in the initializer, just as it's done in this line, so that we know both contracts share the same value, (check that it's non-zero at that time time).

Reply:
I agree. WETH was initially in just one place so I tried to keep to the existing architecture but now its everywhere.
Changing `NodeDelegator` to use an immutable WETH was pretty easy as I have the deployments and upgrades abstracted in helper libraries. ie `NodeDelegatorLib`

Reply:
Fantastically better.